torch/jit/frontend.py

```diff
@@ -286,14 +287,31 @@ def build_class_def(ctx, py_def, methods, properties, self_name):
 
 
 def build_def(ctx, py_def, type_line, def_name, self_name=None):
+    global _args_and_types  # type: ignore[name-defined]
+
+    _args_and_types = False  # type: ignore[name-defined]
     body = py_def.body
     r = ctx.make_range(py_def.lineno + len(py_def.decorator_list),
                        py_def.col_offset,
                        py_def.col_offset + len("def"))
+
+    # If MonkeyType is installed, get all the consolidated traces for the arguments
+    # and the return types from type_trace_db
+
+    type_trace_db = torch.jit._script.type_trace_db
+    if type_trace_db.trace_records:
+        _args_and_types = type_trace_db.get_args_types(def_name)  # type: ignore[name-defined]
+
     param_list = build_param_list(ctx, py_def.args, self_name)
     return_type = None
     if getattr(py_def, 'returns', None) is not None:
         return_type = build_expr(ctx, py_def.returns)
+    else:
+        if type_trace_db.trace_records:
+            # Get the return type annotation from the _args_and_types
+            # dictionary
+            return_type = Var(Ident(r, _args_and_types["return_type"].pop()))  # type: ignore[name-defined, index]
```

Comment:
I think we want to exclude `return` type, please refer to this section in design doc:
https://www.internalfb.com/intern/fburl/?nopassthru&key=g6af5txk

Reply:
Removed!